tsnet/tsnet_test.go

```diff
@@ -175,6 +176,19 @@ func TestConn(t *testing.T) {
 	if string(got) != want {
 		t.Errorf("got %q, want %q", got, want)
 	}
+
+	_, err = s2.Dial(ctx, "tcp", fmt.Sprintf("%s:8082", s1ip)) // some random port
+	if err != nil {
+		var oe *net.OpError
+		if !errors.As(err, &oe) {
+			t.Fatalf("unexpected error: %v", err)
+		}
+		if oe.Err.Error() != "connection was refused" {
```

Comment:
string compare is kinda gross. Windows might have a different message? Compare against the underlying syscall err numbers if possible?

Reply:
sadly the underlying error is just a `*errors.errorString`

```
oe.Err is of type *errors.errorString: "connection was refused"; is syscall.ECONNREFUSED = false
```

Reply:
Dropped the string comparison and just checking for errors (verified test fails w/o the change)